### PR TITLE
Doccument how to create a NamedDomainObjectContainer in the javadocs

### PR DESCRIPTION
**Pull request by JLLeitschuh**
### Context

The  `NamedDomainObjectContainer` doesn't provide any indication how to actually create an instance of itself. I spent a solid 15 min digging around trying to figure out how to create an instance for my plugin.

This should help authors figure out how to create their own instances more quickly.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/api/NamedDomainObjectContainer.java [hunk 1]
```diff
@@ -28,6 +28,8 @@
  * <p>Note that a container is an implementation of {@link java.util.SortedSet}, which means that the container is guaranteed
  * to only contain elements with unique names within this container. Furthermore, items are ordered by their name.</p>
  *
+ * @apiNote Implementations can be created using {@link Project#container} or it's variants.
```
Comment (big-guy) [inline subprojects/core-api/src/main/java/org/gradle/api/NamedDomainObjectContainer.java]: I think the version of javadoc we use doesn't support this tag.  Could you switch this to `@see` and link directly to `Project#container(Class)` and `Project#container(Class,NamedDomainObjectFactory)`?

Comment (JLLeitschuh) [reply subprojects/core-api/src/main/java/org/gradle/api/NamedDomainObjectContainer.java]: Sure, I wasn't sure if you actually wanted this to show up in the generated docs.
I'll change it.

Do you not want me to say "Implementations can be created using" somehow or just drop that?

Comment (big-guy) [reply subprojects/core-api/src/main/java/org/gradle/api/NamedDomainObjectContainer.java]: Sort of, I was thinking you could do something like:
> @see Project#container(Class) Creating a container
> @see Project#container(Class, NamedDomainObjectFactory) Creating a container with a custom factory

And that would render as clickable links.

Comment (JLLeitschuh) [reply subprojects/core-api/src/main/java/org/gradle/api/NamedDomainObjectContainer.java]: Will do.

**Comment by JLLeitschuh** [comment]
@big-guy Anything else required on this?

**Comment by big-guy** [comment]
This is good thanks!

**Pull request merged**